trading_calendars/exchange_calendar_xmos.py

```diff
@@ -0,0 +1,304 @@
+    "New Year's Holiday",
+    month=1,
+    day=5,
+    start_date='2005',
+    end_date='2012',
+)
+NewYearsHoliday5 = Holiday(
+    "New Year's Holiday",
+    month=1,
+    day=6,
+    start_date='2005',
+    end_date='2012',
+)
+
+OrthodoxChristmas = Holiday(
+    'Orthodox Christmas',
+    month=1,
+    day=7,
+    observance=orthodox_christmas_observance,
+)
+
+DefenderOfTheFatherlandDay = Holiday(
+    'Defender of the Fatherland Day',
+    month=2,
+    day=23,
+    observance=defender_of_fatherland_observance,
+)
+
+WomensDay = Holiday(
+    "Women's Day",
+    month=3,
+    day=8,
+    observance=weekend_to_monday,
+)
+
+LabourDay = european_labour_day(observance=weekend_to_monday)
+
+VictoryDay = Holiday(
+    'Victory Day',
+    month=5,
+    day=9,
+    observance=weekend_to_monday,
+)
+
+DayOfRussia = Holiday(
+    'Day of Russia',
+    month=6,
+    day=12,
+    observance=weekend_to_monday,
+)
+
+UnityDay = Holiday(
+    'Unity Day',
+    month=11,
+    day=4,
+    observance=weekend_to_monday,
+    start_date='2005',
+)
+
+NewYearsEve = new_years_eve(
+    observance=new_years_eve_observance,
+    days_of_week=WEEKDAYS,
+)
+
+
+# Adhoc Holidays
+# --------------
+
+new_years_extensions = [
+    pd.Timestamp('2003-01-03', tz=UTC),
+    pd.Timestamp('2013-01-03', tz=UTC),
+    pd.Timestamp('2013-01-04', tz=UTC),
+    pd.Timestamp('2014-01-03', tz=UTC),
+]
+
+orthodox_christmas_extensions = [
+    pd.Timestamp('2003-01-06', tz=UTC),
+    pd.Timestamp('2005-01-10', tz=UTC),
+    pd.Timestamp('2008-01-08', tz=UTC),
+    pd.Timestamp('2009-01-08', tz=UTC),
+    pd.Timestamp('2009-01-09', tz=UTC),
+    pd.Timestamp('2010-01-08', tz=UTC),
+    pd.Timestamp('2011-01-10', tz=UTC),
+    pd.Timestamp('2016-01-08', tz=UTC),
+]
+
+defender_of_the_fatherland_extensions = [
+    pd.Timestamp('2006-02-24', tz=UTC),
+    pd.Timestamp('2010-02-22', tz=UTC),
+]
+
+womens_day_extensions = [
+    pd.Timestamp('2005-03-07', tz=UTC),
+    pd.Timestamp('2011-03-07', tz=UTC),
+    pd.Timestamp('2012-03-09', tz=UTC),
+]
+
+labour_day_extensions = [
+    pd.Timestamp('2002-05-02', tz=UTC),
+    pd.Timestamp('2002-05-03', tz=UTC),
+    pd.Timestamp('2003-05-02', tz=UTC),
+    pd.Timestamp('2004-05-04', tz=UTC),
+    pd.Timestamp('2007-04-30', tz=UTC),
+    pd.Timestamp('2008-05-02', tz=UTC),
+    pd.Timestamp('2012-04-30', tz=UTC),
+    pd.Timestamp('2015-05-04', tz=UTC),
+    pd.Timestamp('2016-05-03', tz=UTC),
+]
+
+victory_day_extensions = [
+    pd.Timestamp('2002-05-10', tz=UTC),
+    pd.Timestamp('2005-05-10', tz=UTC),
+    pd.Timestamp('2006-05-08', tz=UTC),
+    pd.Timestamp('2017-05-08', tz=UTC),
+]
+
+day_of_russia_extensions = [
+    pd.Timestamp('2003-06-13', tz=UTC),
+    pd.Timestamp('2007-06-11', tz=UTC),
+    pd.Timestamp('2008-06-13', tz=UTC),
+    pd.Timestamp('2012-06-11', tz=UTC),
+    pd.Timestamp('2014-06-13', tz=UTC),
+]
+
+unity_day_extensions = [
+    pd.Timestamp('2008-11-03', tz=UTC),
+    pd.Timestamp('2010-11-05', tz=UTC),
+]
```

Comment:
Are all of these "bridge holidays"?  And do the bridge holidays happen every year?  If so, you could also define a bridge holiday by, for example, defining a holiday one day after the real holiday that is only observed on Friday (with a custom `only_friday` observance function).

However, from a quick scan of these dates it looks like these bridge holidays might not be super consistent every year so maybe defining another Holiday wouldn't really work.

Reply:
> you could also define a bridge holiday by...

I did think of this (yes these are all bridge holidays), but from what I can tell these observances are not consistent every year (for example, Day of Russia is on a Tuesday in 2018 but the Monday before is not a holiday). So if I were to make a rule out of the bridge days I'd likely end up with just as many exceptions to that rule as adhoc days here, which would add yet another layer of confusion. My concern though is if that _is_ the rule in future years we will be missing coverage without manually revisiting this calendar every year. Either way you slice it though there's potential for incorrect coverage here, so I will add a comment to note all of this. @jmccorriston feel free to weigh in here, but for now I'm going to leave these as adhoc.

Reply:
Yeah, in that case I think you're right that introducing more Holidays would just lead to more exceptions and we will have to revisit every year anyways, so ad-hoc seems like the way to go.